tests/e2e/tests/istioctl/istioctl_test.go

```diff
@@ -0,0 +1,355 @@
+}
+
+func (t *testConfig) Teardown() error {
+	return nil
+}
+
+func waitForRule(ruleKey string, retries int, durBetweenAttempts time.Duration) error {
+
+	for retry := 0; retry < retries; retry++ {
+		time.Sleep(durBetweenAttempts)
+		var output string
+		var err error
+		if output, err = tc.Kube.Istioctl.RunWithOutput([]string{"get", "route-rules"},
+			"", "", false); err != nil {
+			return err
+		}
+		if strings.Contains(output, ruleKey) {
+			return nil
+		}
+	}
+
+	return fmt.Errorf("created rule did not exist after %v", time.Duration(retries)*durBetweenAttempts)
+}
+
+func waitForRuleRemoval(ruleKey string, retries int, durBetweenAttempts time.Duration) error {
+
+	for retry := 0; retry < retries; retry++ {
+		time.Sleep(durBetweenAttempts)
+		var output string
+		var err error
+		if output, err = tc.Kube.Istioctl.RunWithOutput([]string{"get", "route-rules"},
+			"", "", false); err != nil {
+			return err
+		}
+		if !strings.Contains(output, ruleKey) {
+			return nil
+		}
+	}
+
+	return fmt.Errorf("created rule still exists after %v", time.Duration(retries)*durBetweenAttempts)
+}
+
+// Mixer rules don't have names so we just wait for a rule of the appropriate "kind" (e.g. "quota")
+func waitForMixerRuleType(scope, subject, kind string, retries int, durBetweenAttempts time.Duration) error {
+
+	for retry := 0; retry < retries; retry++ {
+		time.Sleep(durBetweenAttempts)
+		var output string
+		var err error
+		if output, err = tc.Kube.Istioctl.RunWithOutput([]string{"mixer", "rule", "get", scope, subject},
+			"", "", false); err != nil {
+			return err
+		}
+		if strings.Contains(output, kind) {
+			return nil
+		}
+	}
+
+	return fmt.Errorf("created mixer rule kind did not exist after %v", time.Duration(retries)*durBetweenAttempts)
+}
+
+func rulePath(ruleFile string) string {
+	return filepath.Join(tc.rulesDir, ruleFile)
+}
+
+// TestInvalidRuleDetected verifies attempting to create invalid rule fails and generates an error message explaining the problem
+func TestInvalidRuleDetected(t *testing.T) {
+	invalidRules := []ruleFileAndCreateResponse{
+		{
+			ruleFile: "route-rule-101.yaml",
+			response: regexp.MustCompile("percent invalid:  must be in range 0..100"),
+		},
+	}
+
+	for _, invalidRule := range invalidRules {
+		output, err := tc.Kube.Istioctl.RunWithOutput([]string{"create", "--file", rulePath(invalidRule.ruleFile)},
+			"", "", true)
+		if err == nil {
+			glog.Error(fmt.Sprintf("Expected failure but got success %v", output))
+			t.Fatal(fmt.Sprintf("Expected failure but got success %v", output))
+		}
+		if !invalidRule.response.Match([]byte(output)) {
+			t.Errorf("create file %v did not match %v.  Output: %v", invalidRule.ruleFile, invalidRule.response, output)
+		}
+	}
+}
+
+// TestCRUD tests Create, Read, Update, Delete operations for a rule by performing them and scraping the output.
+// Note: If you are improving the output of istioctl, you will have to modify the regexps here used to detect expected output.
```

Comment:
How does one update `istioctl` then without breaking this test since they are in different repos? Maybe this test should be in istio.io/manager unless it depends on non-manager components?